RespokeSDK/NSString+urlencode.m

```diff
@@ -0,0 +1,39 @@
+//
+//  NSString+urlencode.m
+//  Respoke SDK
+//
+//  Copyright 2015, Digium, Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under The MIT License found in the
+//  LICENSE file in the root directory of this source tree.
+//
+//  For all details and documentation:  https://www.respoke.io
+//
+
+#import "NSString+urlencode.h"
+
+@implementation NSString (NSString_Extended)
+
+- (NSString *)urlencode {
```

Comment:
everything I can tell from reading on the internet is that it doesn't do it *properly*. It misses some important characters like "&" and "/". I'll add a comment about the source of the code. All it does is encode everything that isn't in the small subset of allowed characters [a-zA-z0-9\.\-_~] and it replaces ' ' (space) with '_'. 
http://stackoverflow.com/a/8088484/355743